### PR TITLE
Fast boolean evaluation

### PR DESCRIPTION
**Pull request by shromonag**
test_boolean.py has the same set of tests as the boolean one, but its not using unittest for now, I was comparing the results manually.

fastboolean_eval.py has the other boolean evaluation

### CODE REVIEW

File: stl/fastboolean_eval.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    return sat_comp
+    #return lambda x, t, val: [(val << 1) | (x[stl.id][deltat] == True) for deltat in t] 
+
+
+@pointwise_sat.register(stl.LinEq)
+def _(stl):
+    op = op_lookup[stl.op]
+    def sat_comp(x, t):
+        val = 0
+        for tau in t:
```
Comment (mvcisback) [inline stl/fastboolean_eval.py]: Hmm, this fold pattern seems to be very common, in that you are keeping val around just to do the bitwise assignment.

Does gmpy or numpy support doing something like this directly?

File: stl/fastboolean_eval.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
+# TODO: figure out how to deduplicate this with robustness
+# - Abstract as working on distributive lattice
+
+from functools import singledispatch
+import operator as op
+
+import numpy as np
+import sympy as smp
+from lenses import lens
+import gmpy2 as gp
```
Comment (mvcisback) [inline stl/fastboolean_eval.py]: Looking through pypi and googling around, `bitarray` seems to be a nice library for working with bitarrays.
https://pypi.python.org/pypi/bitarray

Aside from `popcount` (which I later argue may not really be helping?) I'm not sure what gmpy gives us.

**Comment by shromonag** [comment]
Oh shit! I realised I pushed the wrong version! I had a cleaned out
version!

**Comment by shromonag** [comment]
Hey, made changes to fast eval. It takes about half the time as the pointwise sat in boolean eval for the pareto curve computation we were looking at.

**Pull request merged**